airbyte-cdk/python/airbyte_cdk/sources/streams/concurrent/cursor.py

```diff
@@ -0,0 +1,153 @@
+import functools
+from abc import ABC, abstractmethod
+from typing import Any, List, Mapping, Optional, Protocol, Tuple
+
+from airbyte_cdk.sources.connector_state_manager import ConnectorStateManager
+from airbyte_cdk.sources.message import MessageRepository
+from airbyte_cdk.sources.streams.concurrent.partitions.partition import Partition
+from airbyte_cdk.sources.streams.concurrent.partitions.record import Record
+
+
+def _extract_value(mapping: Mapping[str, Any], path: List[str]) -> Any:
+    return functools.reduce(lambda a, b: a[b], path, mapping)
+
+
+class Comparable(Protocol):
+    """Protocol for annotating comparable types."""
+
+    @abstractmethod
+    def __lt__(self: "Comparable", other: "Comparable") -> bool:
+        pass
+
+
+class CursorField:
+    def __init__(self, path: List[str]) -> None:
```

Comment:
this should only be `str` as we're phasing out nested cursors. Another good "breaking" change to add to the readme

Reply:
I think I'm missing some context here: Why are we removing this? What happens if a cursor field is nested?

Reply:
nested cursors and primary keys aren't supported by destination v2 https://www.notion.so/Airbyte-Data-Warehouse-Destination-V2-Philosophy-e6081edb87084999948acdc8b9607ea1

> Sources must have their primary key(s) and cursor as top-level properties, as only top-level properties will become ‘real’ columns within the destination.  This is required for typing and deduplication.  Ideally, sources will also place any ‘important’ (as judged by the source developer) properties at the top level as well.

They're still valid as per the airbyte protocol, but I'd prefer we keep the new interface simpler and "in line with the best practices" when possible